src/components/ChatRoomPage/MessageList.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState, useEffect, useRef } from 'react';
+import profileIcon from "../../assets/ChatRoom/profile.svg";
+import { formatTime } from '../../utils/ClockUtils';
+
+interface User {
+    userId: number;
+    userName: string;
+}
+
+interface Message {
+    senderId: number;
+    text: string;
+    timestamp: Date;
+}
+
+interface MessageListProps {
+    messages: Message[];
+    currentUserId: number;
+    users: User[];
+}
+
+const groupMessages = (messages: Message[]) => {
+    return messages.reduce((arr, message, index) => {
+        const timeKey = formatTime(message.timestamp);
+        const prevMessage = messages[index - 1];
+        const isSameSender = prevMessage && prevMessage.senderId === message.senderId;
+        const isSameMinute = prevMessage && formatTime(prevMessage.timestamp) === timeKey;
+
+        if (!isSameSender || !isSameMinute) {
+            arr.push({ timeKey, senderId: message.senderId, messages: [message] });
+        } else {
+            arr[arr.length - 1].messages.push(message);
+        }
+
+        return arr;
+    }, [] as { timeKey: string; senderId: number; messages: Message[] }[]);
+};
+
+const MessageList: React.FC<MessageListProps> = ({ currentUserId, messages, users }) => {
+    const scrollRef = useRef<HTMLDivElement>(null);
+
+    // senderId로 사용자 이름 찾기
+    const getUserName = (senderId: number) => {
+        const user = users.find(user => user.userId === senderId);
+        return user ? user.userName : '(알 수 없음)';
+    };
+
+    useEffect(() => {
+        if (scrollRef.current) {
+            scrollRef.current.scrollIntoView({ behavior: 'smooth' });
+        }
+    }, [messages]);
+
+    const groupedMessages = groupMessages(messages);

```

Comment:
```suggestion
  const groupedMessages = useMemo(() => groupMessages(messages), [messages]);

```

messages가 변경될 때마다 groupMessages가 다시 계산되는 것을 방지하고, 성능을 최적화하기 위해 useMemo를 사용하는 건 어떨까요?